SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java

```diff
@@ -142,21 +158,42 @@ public MailTemplateHelpers(SandboxManager sandboxManager, SingularityConfigurati
     }
   }
 
-  private Optional<Long> getMaybeOffset(String slaveHostname, String fullPath, String filename, Optional<String> directory) {
+  // Searches through the file,
+  private Optional<Long> getMaybeOffset(final String slaveHostname, final String fullPath, final Long logLength, Optional<SingularityTask> task) {
+    long offset = 0;
+    String regex = getLogErrorRegex(task);
+    long length = logLength + regex.length(); // Get extra so that we can be sure to find the error
+    Pattern pattern = Pattern.compile(regex);
     Optional<MesosFileChunkObject> logChunkObject;
-    try {
-      logChunkObject = sandboxManager.read(slaveHostname, fullPath, Optional.<Long>absent(), Optional.<Long>absent());
-    } catch (RuntimeException e) {
-      LOG.error("Sandboxmanager failed to read {}/{} on slave {}", directory.get(), filename, slaveHostname, e);
-      return Optional.absent();
-    }
-    if (logChunkObject.isPresent()) {
-      return Optional.of(logChunkObject.get().getOffset());
-    } else {
-      LOG.error("Failed to get offset for log file {}", fullPath);
-      return Optional.absent();
+    Optional<MesosFileChunkObject> previous = Optional.absent();
+
+    while (true) { //Continue until we reach the end of the file, at which point we return from within the while loop
```

Comment:
Two things for this loop:
- In the case that the file is massive (for example, a wrongly configured service running for a few hours can spew GBs upon GBs of logs into a file), should we limit the amount of log file we will search?
- `while true`  always makes me a bit uneasy that we could be stuck here forever. Maybe we should check the ending offset of the log before starting the loop, so that even if something goes wrong here, we can still have a defined end and no chance of ever getting stuck here?